pkg/firewall/reconciler.go

```diff
@@ -0,0 +1,89 @@
+package firewall
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/capg-firewall-rule-operator/pkg/cidr"
+)
+
+const AnnotationBastionAllowListSubnets = "bastion.gcp.giantswarm.io/allowlist"
+
+//counterfeiter:generate . FirewallsClient
+type FirewallsClient interface {
+	ApplyRule(context.Context, *capg.GCPCluster, Rule) error
+	DeleteRule(context.Context, *capg.GCPCluster, string) error
+}
+
+func NewRuleReconciler(
+	defaultBastionHostAllowList []string,
+	firewallClient FirewallsClient,
+) *RuleReconciler {
+	return &RuleReconciler{
+		defaultBastionHostAllowList: defaultBastionHostAllowList,
+		firewallClient:              firewallClient,
+	}
+}
+
+type RuleReconciler struct {
+	defaultBastionHostAllowList []string
+
+	firewallClient FirewallsClient
+}
+
+func (r *RuleReconciler) Reconcile(ctx context.Context, cluster *capg.GCPCluster) error {
+	logger := r.getLogger(ctx)
+
+	ruleName := getBastionFirewallRuleName(cluster.Name)
+	tagName := fmt.Sprintf("%s-bastion", cluster.Name)
```

Comment:
My OCD kicked in seeing a function for `ruleName` but an inlined value for `tagName` :smile: Any reason not to inline both (or use function for both)?

Reply:
I think it was because the rule name is needed for both create and delete, while the tag is only for create. But I'll add a function for the tag too